sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -62,6 +64,11 @@ object DateTimeUtils {
 
   @transient lazy val defaultTimeZone = TimeZone.getDefault
 
+  // These are ISO 8601 compliant `FastDateFormat`s. These are thread-safe and compatible
+  // with `SimpleDateFormat`.
+  lazy val iso8601TimeFormat = FastDateFormat.getInstance("yyyy-MM-dd'T'HH:mm:ss.SSSZZ")
```

Comment:
Yup, that's why I avoided to use `SimpleDateFormat`. I will address the other comments now. Thank you.